### PR TITLE
Basic docs for jacobian +=

### PR DESCRIPTION
**Pull request by WardBrian**
#### Submission Checklist

- [x] Builds locally
- [x] New functions marked with `` <<{ since VERSION }>>``
- [x] Declare copyright holder and open-source license: see below

#### Summary

Docs for https://github.com/stan-dev/stanc3/pull/1435

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):

Simons Foundation

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC BY-ND 4.0 (https://creativecommons.org/licenses/by-nd/4.0/)

### CODE REVIEW

File: src/reference-manual/statements.qmd [hunk 1]
```diff
@@ -361,6 +361,38 @@ including arrays of vectors and matrices.   For container arguments,
...
+## Increment log density with a change of variables adjustment
+
+
+A variant of the `target +=` statement described above is the
+`jacobian +=` statement. This can be used in the transformed parameters block
+or in functions ending with `_jacobian` to mimic the log Jacobian
+adjustments accrued by built-in variable transforms.
+
+Similarly to those implemented for the built-in transforms, these Jacobian adjustment
+may be skipped for optimization.
```
Comment (bob-carpenter) [inline src/reference-manual/statements.qmd:373]: I would say "may be skipped for maximum likelihood estimation via optimization" to make it clear the optimization isn't speed related here.  If you leave the transforms and optimize, you get the MAP (max a posteriori) estimate.

File: src/reference-manual/statements.qmd [hunk 2]
```diff
@@ -361,6 +361,38 @@ including arrays of vectors and matrices.   For container arguments,
...
+
+For example, here is a program which re-creates the existing
```
Comment (bob-carpenter) [inline src/reference-manual/statements.qmd:375]: "recreate" isn't hyphenated

File: src/reference-manual/statements.qmd [hunk 3]
```diff
@@ -361,6 +361,38 @@ including arrays of vectors and matrices.   For container arguments,
...
+}
+data {
+  real ub;
+}
+parameters {
+  real b_raw;
+}
+transformed parameters {
+  real b = upper_bound_jacobian(b_raw, ub);
+}
```
Comment (bob-carpenter) [inline src/reference-manual/statements.qmd:393]: I think you want to go further and illustrate a positive-constrained distribution on `b`, such as

```stan
model {
  b ~ lognormal(0, 1);
}
```

File: src/reference-manual/statements.qmd [hunk 4]
```diff
@@ -361,6 +361,38 @@ including arrays of vectors and matrices.   For container arguments,
...
+```
+
 ### Accessing the log density {-}
 
 To access accumulated log density up to the current execution point,
```
Comment (bob-carpenter) [inline src/reference-manual/statements.qmd:398]: (I realize this isn't yours, but it should be "the accumulated"

File: src/stan-users-guide/user-functions.qmd [hunk 5]
```diff
@@ -287,6 +287,32 @@ transformed parameters {
...
+data {
+  real ub;
+}
+parameters {
+  real b_raw;
+}
+transformed parameters {
+  real b = upper_bound_jacobian(b_raw, ub);
+}
+```
```
Comment (bob-carpenter) [inline src/stan-users-guide/user-functions.qmd:315]: This is a duplicate and I'd like a sampling statement for `b` in the model block.

File: src/stan-users-guide/using-stanc.qmd [hunk 6]
```diff
@@ -647,7 +647,9 @@ Warning:
     Left-hand side of distribution statement (~) may contain a non-linear
     transform of a parameter or local variable. If it does, you need
     to include a target += statement with the log absolute determinant
-    of the Jacobian of the transform.
+    of the Jacobian of the transform. You could also consider defining
```
Comment (bob-carpenter) [inline src/stan-users-guide/using-stanc.qmd:650]: You could also consider defining -> Alternatively, you could define
using -> use

**Review by bob-carpenter — CHANGES_REQUESTED** [review]
Minor corrections.

File: src/stan-users-guide/using-stanc.qmd (hunk 6 above)
Comment (WardBrian) [reply src/stan-users-guide/using-stanc.qmd:650]: This is a copy of a warning message stanc3 emits in pedantic mode, so it's wording would need to be changed there

File: src/reference-manual/statements.qmd (hunk 3 above)
Comment (WardBrian) [reply src/reference-manual/statements.qmd:393]: That is no longer "just" a recreation of the built in transform, so I've added it but commented out as an example

File: src/stan-users-guide/user-functions.qmd (hunk 5 above)
Comment (WardBrian) [reply src/stan-users-guide/user-functions.qmd:315]: Duplication is intentional here as it's in a very different part of the docs, but I can do something unique if you prefer

File: src/reference-manual/statements.qmd (hunk 3 above)
Comment (bob-carpenter) [reply src/reference-manual/statements.qmd:393]: Good point.  Then it probably makes more sense to add it as a separate comment to say that you can then use a distribution on the transformed parameter.

File: src/stan-users-guide/user-functions.qmd (hunk 5 above)
Comment (bob-carpenter) [reply src/stan-users-guide/user-functions.qmd:315]: Understood.  I'd change it up just for variety, but I can see the motivation for duplicating, so up to you.

File: src/stan-users-guide/using-stanc.qmd (hunk 6 above)
Comment (bob-carpenter) [reply src/stan-users-guide/using-stanc.qmd:650]: Aha.  Our wording's not so great and if we change it, all hell breaks loose on the R side given their goofy unit testing in other packages.

**Pull request merged**